src/containers/User/UserArea.tsx

```diff
@@ -446,63 +459,125 @@ const getNewDropdownsState = (basepath: string, dropdownsState: DropdownsState):
-        admin: { active: false, selection: '' },
+        config: { active: false, selection: '' },
         intRefDocs: { active: false },
         extRefDocs: { active: false },
       }
     case 'admin':
       return {
         dashboard: { active: false },
-        templates: { active: false, selection: '' },
+        applicationList: { active: false, selection: '' },
         dataViews: { active: false, selection: '' },
         manage: { active: false, selection: '' },
-        admin: { active: true, selection: dropdownsState.admin.selection },
+        config: { active: true, selection: dropdownsState.config.selection },
         intRefDocs: { active: false },
         extRefDocs: { active: false },
       }
     // Don't need one for ref docs because they open in new tab
     default:
       return {
         dashboard: { active: false },
-        templates: { active: false, selection: '' },
+        applicationList: { active: false, selection: '' },
         dataViews: { active: false, selection: '' },
         manage: { active: false, selection: '' },
-        admin: { active: false, selection: '' },
+        config: { active: false, selection: '' },
         intRefDocs: { active: false },
         extRefDocs: { active: false },
       }
   }
 }
+
+interface StandardMenuOption {
+  key: string
+  text: string
+  value: string
+}
+
+interface NestedMenuOption {
+  key: string
+  text: string
+  content: JSX.Element
+}
+
+interface MenuInput<T> {
+  filterMethod: (e: T) => boolean
+  mapMethod: (e: T) => StandardMenuOption
+  getSubmenuMethod: (e: T) => string | null
+  onClick: (value: string) => void
+}
+
+function constructNestedMenuOptions<T>(
+  items: T[],
+  { filterMethod, mapMethod, getSubmenuMethod, onClick }: MenuInput<T>
+): (StandardMenuOption | NestedMenuOption)[] {
+  const filteredItems = items.filter(filterMethod)
+  const menus: (StandardMenuOption | NestedMenuOption)[] = []
+  const subMenus: Record<string, StandardMenuOption[]> = {}
+  filteredItems.forEach((item) => {
+    const submenu = getSubmenuMethod(item)
+    if (!submenu) menus.push(mapMethod(item))
+    else {
+      if (submenu in subMenus) subMenus[submenu].push(mapMethod(item))
+      else subMenus[submenu] = [mapMethod(item)]
+    }
+  })
+
+  Object.entries(subMenus).forEach(([submenu, items]) => {
+    menus.push({
+      key: submenu,
+      text: submenu,
+      content: (
+        <Dropdown item text={submenu}>
+          <Dropdown.Menu style={{ transform: 'translateY(-25px)' }}>
+            {items.map((item) => {
+              const { key, text, value } = item
+              return (
+                <Dropdown.Item
+                  key={key}
+                  text={text}
+                  value={value}
+                  onClick={(_, data) => onClick(data.value as string)}
+                />
+              )
+            })}
+          </Dropdown.Menu>
+        </Dropdown>
+      ),
+    })
+  })
+
+  return menus
+}
```

Comment:
This is the main addition to this issue, the rest is basically tidying up and small refactors.